pravega-k8-log-collection/pravega-k8s-log.sh

```diff
@@ -0,0 +1,34 @@
+#/usr/bin/env bash
+set -euo pipefail
+mkdir -p /tmp/logdir
+output_dir="/tmp/logdir"
+kubectl get po,svc,pvc --all-namespaces > $output_dir/kubectl-get-po-svc-pvc-all-namespace_output
+IFS=$'\n'
+for i in $(cat $output_dir/kubectl-get-po-svc-pvc-all-namespace_output | grep -i pravega)
+do
+echo $i
+if [[ $i == *"po/"* ]] || [[ $i == *"pod/"* ]]; then
+    if [ "`echo $i | awk '{print $4}'`" == "Running" ]; then
```

Comment:
Can we only collect logs of running pods? What about pods that have terminated (successfully)?

Reply:
There are couple of reason behind this,
1. Log collection depends on kubectl, it interact with that pod and run log streaming (for recent and/or previous instances (if exist)) or copy complete logs (older rotated) from the pod log directory (/opt/pravega/logs). Both the cases if pod is in "not running" state these commands fails.
2. For Pravega deployment we use Replicaset(controller) and Statefulset(Zk, Bookie, Segmentstore), which means if there would be termination it will try restart and maintain state to be running.
3. Successful/un-successful terminated pod logs are getting collected with previous instance logging or complete copy.
4. There are other phases in pod deployment (like "Unknown": For some reason the state of the Pod could not be obtained, "Evicted": due to some resource issue pod terminated), - script failure will be observed if we wouldn't include "running" pod detection in the script.